api.go

```diff
@@ -207,7 +219,18 @@ func (r *request) toHTTP() (*http.Request, error) {
 	}
 
 	// Create the HTTP request
-	return http.NewRequest(r.method, urlRaw, r.body)
+	req, err := http.NewRequest(r.method, urlRaw, r.body)
+
+	// Setup auth
+	if err == nil && r.config.HttpAuth != nil {
+		r.config.HttpAuth.auth(req)
+	}
+
+	return req, err
+}
+
+func (a *HttpBasicAuth) auth(req *http.Request) {
```

Comment:
Yes, why not. Just though to hide details of auth mechanism inducted by implementation of particular *HttpAuth in a case a) auth impl will differ b) http client api impl will change. But in that particular case this is over complication. So, should i do it tomorrow?